src/api/actions/auth/authActions.ts

```diff
@@ -3,9 +3,9 @@ import { InfiniteQueryFn } from 'hooks/useInfiniteQuery/useInfiniteQuery.types';
 
 import { GetUsersResponse, LoginMutationArguments, LoginMutationResponse } from './authActions.types';
 
-export const meQueryKey = '/me';
-export const usersQueryKey = 'users';
-export const loginQueryKey = 'login';
+export const meQueryKey = ['/me'];
```

Comment:
I hate those arrays but I do understand that we need this because of a change in react-query.

Maybe we could make a tuple creator for this https://tanstack.com/query/v4/docs/guides/query-keys
so that it takes 3 arguments, key string, url parameters [string] and query parameters {} and produces a typed tuple? Arrays returned by a function will automatically get inferred to tuple so it's a low hanging fruit

Reply:
Sounds good, im only worried that this will get too complicated because we use query key as url